src/lib.rs

```diff
@@ -158,6 +198,96 @@ impl Primitive {
                 }
                 written
             },
+            Primitive::U8LE => {
```

Comment:
If you interpret file as individual bytes rather than whole array there is no point in this variant.
u8 is only single byte and endianness is irrelevant to single byte

Reply:
Yes, I wanted to have the api be uniform. So I think we should keep this variant, even though it is equivalent to u8

Reply:
Ok, then just move it to `Primitive::U8` branch and match it all to the same branch
There is no point in useless code